pkg/controller/provider/controller.go

```diff
@@ -196,11 +198,42 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 			deployment)
 
 		// If the deployment does not exist
-		if k8serr.IsNotFound(err) {
-			r.CreateOVAServerDeployment(provider, ctx)
-		} else if err != nil {
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				err = r.CreateOVAServerDeployment(provider, ctx)
+				if err != nil {
+					r.handleServerCreationFailure(provider, err)
+					return
+				}
+				provider.Status.Phase = Initializing
```

Comment:
can't we reuse the `Staging` phase and check `deployment.Status.AvailableReplicas` as part of the "test connection" phase?

Reply:
same here, staging should be a different phase for the creation of the inventory, here we first only setting up the sever and everything related to it before, so I think it should be a separate phase as well (also more easy for the debug in case any errors happened)